pre_award/assess/assessments/routes.py

```diff
@@ -1311,25 +1311,30 @@ def request_changes(application_id, sub_criteria_id, theme_id):
     assessment_status = determine_assessment_status(sub_criteria.workflow_status, state.is_qa_complete)
     theme_answers_response = get_sub_criteria_theme_answers_all(application_id, theme_id)
 
-    form = RequestChangesForm(
-        question_choices=[(question["field_id"], question["question"]) for question in theme_answers_response]
-    )
+    question_choices = [
+        (question["field_id"], question["question"], question.get("answer", "")) for question in theme_answers_response
+    ]
 
+    form = build_request_changes_form(question_choices)()
+    justification_data = {field_id: getattr(form, f"reason_{field_id}").data for field_id, _, _ in question_choices}
     if request.method == "POST" and form.validate_on_submit():
-        submit_change_request(
-            application_id=application_id,
-            flag_type=FlagType.RAISED.name,
-            user_id=g.account_id,
-            justification=form.justification.data,
-            field_ids=form.field_ids.data,
-            section=[sub_criteria_id],
-            is_change_request=True,
-        )
+        for field_id, justification in justification_data.items():
+            justification = justification.strip()
```

Comment:
Good practice is to not modify objects/dictionary properties, as it can lead to unexpected behaviour later in your code. I'd create a separate variable, or in this case no bother to store it and do a quick return

You can do a quick return:
```python
if not justification.strip():
    continue
```